src/middleware.ts

```diff
@@ -1,10 +1,47 @@
 import { withAuth } from "next-auth/middleware";
+import { NextResponse } from 'next/server'
+import type { NextRequest } from 'next/server'
 
 export const pagesNextAuthOptions = {
   signIn: "/signin",
 };
 
+// Custom middleware
+export function middleware(req: NextRequest) {
+
+  // Checking `/api` routes
+  if (req.nextUrl.pathname.startsWith("/api")) {
+
+    const authHeader = req.headers.get("authorization");
+
+    if (!authHeader ||
```

Comment:
did you test if this properly takes the middleware from sphere? I think it will fail bc the authorization does not include bearer